lighty-examples/lighty-gnmi-community-restconf-app/src/test/java/io/lighty/examples/controllers/gnmi/GnmiRestconfExampleTest.java

```diff
@@ -0,0 +1,312 @@
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class GnmiRestconfExampleTest {
+
+    private static final Duration REQUEST_TIMEOUT_DURATION = Duration.ofMillis(10_000L);
+    private static final Duration POLL_INTERVAL_DURATION = Duration.ofMillis(1_000L);
+    private static final Duration WAIT_TIME_DURATION = Duration.ofMillis(10_000L);
+    private static final String CA_CERTIFICATE_PATH = "src/main/assembly/resources/certificates/ca.crt";
+    private static final String CLIENT_CERTIFICATE_PATH = "src/main/assembly/resources/certificates/client.crt";
+    private static final String CLIENT_KEY_PATH = "src/main/assembly/resources/certificates/client.key";
+
+    private static final String DEVICE_ID = "gnmi-simulator";
+    private static final int DEVICE_PORT = 3333;
+    private static final String DEVICE_IP = "127.0.0.1";
+    private static final String KEYSTORE_ID = "keystore-id-1";
+    private static final String USERNAME = "Admin";
+    private static final String PASSWORD = "Admin";
+    private static final String SYSTEM_AUTHENTICATION = "openconfig-system:authentication";
+    private static final String STATE = "state";
+    private static final String CONFIG = "config";
+    private static final String AUTHENTICATION_METHOD = "authentication-method";
+
+    private static final String RESTCONF_PATH = "http://localhost:8888/restconf";
+    private static final String ADD_CERTIFICATE_PATH
+            = RESTCONF_PATH + "/operations/gnmi-certificate-storage:add-keystore-certificate";
+    private static final String MOUNTPOINT_PATH
+            = RESTCONF_PATH + "/data/network-topology:network-topology/topology=gnmi-topology/node=" + DEVICE_ID;
+    private static final String MOUNTPOINT_STATUS_PATH = MOUNTPOINT_PATH + "/gnmi-topology:node-state/node-status";
+    private static final String NODE_STATUS_RESPONSE_READY = "{\"gnmi-topology:node-status\":\"READY\"}";
+    private static final String GET_INTERFACES = MOUNTPOINT_PATH + "/yang-ext:mount/openconfig-interfaces:interfaces";
+    private static final String GET_SYSTEM = MOUNTPOINT_PATH + "/yang-ext:mount/openconfig-system:system";
+    private static final String GET_AUTHENTICATION
+            = MOUNTPOINT_PATH + "/yang-ext:mount/openconfig-system:system/aaa/authentication";
+    private static final String GET_AUTHENTICATION_CONFIG = GET_AUTHENTICATION + "/config" ;
+    private static final String GET_CONFIG_AUTHENTICATION_CONFIG = GET_AUTHENTICATION_CONFIG + "?content=config" ;
+    private static final String GET_CONFIG_AUTHENTICATION = GET_AUTHENTICATION + "?content=config";
+
+    private static ExecutorService httpClientExecutor;
+    private static HttpClient httpClient;
+
+    @BeforeAll
+    public static void startUp() throws IOException {
+        httpClientExecutor = Executors.newSingleThreadExecutor();
```

Comment:
these startup, teardown methods and also tests should not be throwing exceptions, handle them appropriately, also in tests